src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala

```diff
@@ -142,8 +142,7 @@ class JdbcReadJournal(config: Config, configPath: String)(implicit val system: E
         case _ =>
           delaySource
             .flatMapConcat { _ =>
-              currentJournalEventsByPersistenceId(persistenceId, from, toSequenceNr)
-                .take(readJournalConfig.maxBufferSize)
+              currentJournalEventsByPersistenceId(persistenceId, from, from + (readJournalConfig.maxBufferSize - 1))
```

Comment:
what if `toSequenceNr` is smaller than `from + (readJournalConfig.maxBufferSize - 1)`

Perhaps we could re-use some of the code used for the eventsByTag query. Where a similar mechanism is being used

Reply:
Good point. I've just updated with a fix for this case